src/sentry/static/sentry/app/views/settings/components/settingsLayout.tsx

```diff
@@ -14,77 +18,182 @@ type Props = {
+  }
+
+  render() {
+    const {params, routes, route, router, renderNavigation, children} = this.props;
+    const {navVisible, navOffsetTop} = this.state;
+
+    // We want child's view's props
+    const childProps =
+      children && React.isValidElement(children) ? children.props : this.props;
+    const childRoutes = childProps.routes || routes || [];
+    const childRoute = childProps.route || route || {};
+    const shouldRenderNavigation = typeof renderNavigation === 'function';
+
+    return (
+      <React.Fragment>
+        <SettingsColumn>
+          <SettingsHeader ref={this.headerRef}>
+            <HeaderContent>
+              {shouldRenderNavigation && (
+                <NavMenuToggle
+                  priority="link"
+                  icon={navVisible ? <IconClose /> : <IconMenu />}
+                  onClick={() => this.toggleNav(!navVisible)}
+                />
+              )}
+              <StyledSettingsBreadcrumb
+                params={params}
+                routes={childRoutes}
+                route={childRoute}
+              />
+              <SettingsSearch routes={routes} router={router} params={params} />
+            </HeaderContent>
+          </SettingsHeader>
+
+          <MaxWidthContainer>
+            {shouldRenderNavigation && (
+              <SidebarWrapper isVisible={navVisible} offsetTop={navOffsetTop}>
+                {renderNavigation!()}
+              </SidebarWrapper>
+            )}
+            <NavMask isVisible={navVisible} onClick={() => this.toggleNav(false)} />
+            <Content>{children}</Content>
+          </MaxWidthContainer>
+        </SettingsColumn>
+      </React.Fragment>
+    );
+  }
 }
 
-SettingsLayout.propTypes = {
-  renderNavigation: PropTypes.func,
-  route: PropTypes.object,
-  router: PropTypes.object,
-  routes: PropTypes.array,
-};
+const SettingsColumn = styled('div')`
+  display: flex;
+  flex-direction: column;
+  flex: 1; /* so this stretches vertically so that footer is fixed at bottom */
+  min-width: 0; /* fixes problem when child content stretches beyond layout width */
+  footer {
+    margin-top: 0;
+  }
+`;
+
+const HeaderContent = styled('div')`
+  display: flex;
+  align-items: center;
+  justify-content: space-between;
+`;
+
+const NavMenuToggle = styled(Button)`
+  display: none;
+  margin-right: ${space(2)};
+  color: ${p => p.theme.gray600};
+  &:hover,
+  &:focus,
+  &:active {
+    color: ${p => p.theme.gray800};
+  }
+  @media (max-width: ${p => p.theme.breakpoints[0]}) {
+    display: block;
+  }
+`;
+
+const StyledSettingsBreadcrumb = styled(SettingsBreadcrumb)`
+  flex: 1;
+`;
 
 const MaxWidthContainer = styled('div')`
   display: flex;
   max-width: ${p => p.theme.settings.containerWidth};
-  min-width: 600px; /* for small screen sizes, we need a min width to make it semi digestible */
   flex: 1;
 `;
 
-const SidebarWrapper = styled('div')`
+const SidebarWrapper = styled('div')<{isVisible: boolean; offsetTop: number}>`
```

Comment:
are you sure that these props are not being added to the dom?

Reply:
They are in camelCase, we should be safe.

Reply:
I would check it anyway.. just in case... this week I used isClipped..and thought the same, but when I checked the dom..it was there 😉